technology-adapter-api.md

```diff
@@ -26,6 +26,62 @@ Message:
 
 val - is error code, src - origin of the error.
 
+## Requesting status of a single device.
+
+An adapter has to support an API for requesting a device state.
+
+### Command
+
+Topic: `pt:j1/mt:cmd/rt:ad/rn:zigbee/ad:1`
+
+Message:
+
+```json
+{
+    "serv": "zigbee",
+    "type": "cmd.network.get_node",
+    "val_t": "int",
+    "val": 80,
+    "props": null,
+    "tags": null,
+    "ctime": "2018-11-22T23:14:40+0100",
+    "uid":"76533455876765"
+}
+```
+
+
+## Report event.
+
+Report events has to be sent by adapters after invocation of command and also autonomously, when device's state change is being detected.
+
+Topic: `pt:j1/mt:cmd/rt:ad/rn:zigbee/ad:1`
+
+Message:
+
+```json
+{
+    "serv": "zigbee",
+    "type": "evt.network.node_report",
+    "val_t": "object",
+    "val": {
+      "address": 80,
+      "status": "UP",
+    },
+    "props": null,
+    "tags": null,
+    "ctime": "2018-11-22T23:14:40+0100",
+    "uid":"76533455876765"
+}
+```
+
+### Definitions
+
+* `status` describes a reachability of a device, well-defined statuses include: `UP` and `DOWN`.
+
+* Command value has to be treated as `node_id` (according to Z-Wave) or `uuid` (according to Zigbee). Above description is also applicable for `address` property within event report value.
```

Comment:
should we maybe have it be a string and not int, so that we can support protocols and adapters having addresses which are not integers?

Reply:
Address should always be a string, 100%.

Reply:
Also please remember that we will want to have similar support eventually in other adapters, including cloud/custom adapters as well as maybe a future matter adapter?

Reply:
ah, yes, also to avoid confusion that address is sometimes a string and sometimes another type, for sure

Reply:
Maybe: "Command value is an `address` uniquely identifying a thing. In case of Z-Wave `address` is equal to `node_id` within the network, while in Zigbee it is equal to `uuid`. Other adapters may have their own understanding of the address, such as unique ID in the third party cloud API."

Reply:
fixed